modules/services/hercules-ci-agent/default.nix

```diff
@@ -0,0 +1,67 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.hercules-ci-agent;
+  user = config.users.users.hercules-ci-agent;
+in
+{
+  imports = [ ./common.nix ];
+
+  options.services.hercules-ci-agent = {
+
+    logFile = mkOption {
+      type = types.path;
+      default = "/var/log/hercules-ci-agent.log";
+      description = "Stdout and sterr of hercules-ci-agent process.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    launchd.daemons.hercules-ci-agent = {
+      script = "exec ${cfg.package}/bin/hercules-ci-agent --config ${cfg.tomlFile}";
+
+      path = [ config.nix.package ];
+      environment = {
+        NIX_SSL_CERT_FILE = "${pkgs.cacert}/etc/ssl/certs/ca-bundle.crt";
+      };
+
+      serviceConfig.KeepAlive = true;
+      serviceConfig.RunAtLoad = true;
+      serviceConfig.StandardErrorPath = cfg.logFile;
+      serviceConfig.StandardOutPath = cfg.logFile;
+      serviceConfig.GroupName = "hercules-ci-agent";
+      serviceConfig.UserName = "hercules-ci-agent";
+      serviceConfig.WorkingDirectory = user.home;
+      serviceConfig.WatchPaths = [
+        cfg.settings.staticSecretsDirectory
+      ];
+    };
+
+    system.activationScripts.preActivation.text = ''
+      touch '${cfg.logFile}'
+      chown ${toString user.uid}:${toString user.gid} '${cfg.logFile}'
+    '';
+
+    # Trusted user allows simplified configuration and better performance
+    # when operating in a cluster.
+    nix.trustedUsers = [ "hercules-ci-agent" ];
+    services.hercules-ci-agent.settings.nixUserIsTrusted = true;
+
+    users.knownGroups = [ "hercules-ci-agent" ];
+    users.knownUsers = [ "hercules-ci-agent" ];
+
+    users.users.hercules-ci-agent = {
+      uid = mkDefault 532;
```

Comment:
I think a uid from a different range would be better, regular users get ids starting with 501 which could result in conflicts quite quickly.

Reply:
This follows the example set by the existing services `hydra` and `ofborg`.
I've looked through the launchd-related man pages but couldn't find any recommendations and my googling wasn't successful either. Do you have a suggestion for a range?

Reply:
Hmm, I guess I made the assumption that there are most likely only a couple regular users on a machine used for CI.
From what I understand of the nixbld discussion role accounts (kind of hidden unix users?) have to be in the 200-300 range, of which the build users will already take up a decent chunk depending on the size of the machine.

Reply:
I'm ok with whatever scheme you decide.
I think you're referencing this PR https://github.com/NixOS/nix/pull/4532 which suggests to use the 300-400 range.
Perhaps counting down (https://github.com/NixOS/nix/pull/4532#issuecomment-774791756), so what about 399?

Reply:
`gid` has the check `check = t: isInt t && t > 501;`. I think we should loosen that to `>= 300`